packages/apollo-server-express/src/ApolloServer.ts

```diff
@@ -29,9 +30,27 @@ export interface ServerRegistration {
   bodyParserConfig?: OptionsJson | boolean;
   onHealthCheck?: (req: express.Request) => Promise<any>;
   disableHealthCheck?: boolean;
-  gui?: boolean;
+  gui?: ((req: express.Request) => Partial<GuiOptions>) | Partial<GuiOptions>;
```

Comment:
I think it's important to retain the option to use or return a boolean for the gui option. Also, we'll have to include this logic across all of the integrations that use a gui. If you think of another way to encapsulate this logic, I'm happy to go for it.